src/OrbitGl/include/OrbitGl/TrackHeader.h

```diff
@@ -36,7 +36,7 @@ class TrackHeader : public CaptureViewElement, public std::enable_shared_from_th
   [[nodiscard]] const TriangleToggle* GetCollapseToggle() const { return collapse_toggle_.get(); }
   [[nodiscard]] TriangleToggle* GetCollapseToggle() { return collapse_toggle_.get(); }
 
-  [[nodiscard]] float GetHeight() const override { return layout_->GetTrackTabHeight(); }
+  [[nodiscard]] float GetHeight() const override { return GetParent()->GetHeight(); }
```

Comment:
That one here is actually the issue that leads to the crash.

Basically `GpuSubmissionTrack::GetHeight` is calling `header_->GetHeight()`, which again with this line here is calling `GpuSubmissionTrack::GetHeight` (as the track is the parent of the header.

Reply:
@reichlfl FYI

Reply:
Adding a quick fix (which should not be the way to go), we see the same issues as in the memory track:
![Screenshot 2022-12-16 at 11 58 03](https://user-images.githubusercontent.com/2725914/208083810-1d1aef38-ba91-4010-b104-bd0a5e17c19e.png)

Reply:
@pierricgimmig This crash here does not seem to be fixed.

Reply:
Ah ok, hopefully I can repro with @ronaldfw 's capture.

Reply:
So, I looked into your fix for the crash. I still think this is against the layouting principles we came up with. The idea about our parent/child relationship is, that the position and size is controlled by the parent (this is mostly due to the lack of flexible layouting), and we tried to be consistent about this.

Now with your change you must remember that you can't call `GetHeight` for you the header, while it is fine and maybe even required to call the same method for your other childs (like for subtracks).

Maybe @reichlfl has a better idea how to implement the `TrackHeader::GetHeight` method.

Reply:
I gave this another thought. The track header is completely inside the track (which is correct). It's relevant height however, is not the complete track's height, but only the height of the label (or the toggle) + some slight spacing at the top and below the label.

So you can implement this method with information locally known here.

This also allows to fix the issues with the header being out of the track in case of subtracks, as we can move the subtracks below the parent's header when the parent track is expanded.

Reply:
I guess the easy fix would be to explicitly set the header's height in `Track::DoUpdateLayout()`.

Alternatively, @reichlfl mentioned that we could do something similar to the capture view element's `width` using the `LayoutFlags`.

But, I guess the simple solution would work for now.

Reply:
@pierricgimmig I'm not sure if this discussion happened before or after you requested the last review - just to be sure this doesn't get lost, as far as I see it this is the main conceptual issue that prevents this from being merged. Children cannot use the parent height to determine their own height, only the other way around. Please instead introduce a `SetHeight` method for the headers, and call this during `Track::DoUpdateLayout`. Make sure to set the "update required" flags accordingly within `SetHeight` (see other layout-changing setters for reference).

Reply:
Implemented your suggestion regarding a SetHeight method @reichlfl